### PR TITLE
Fixing demo docs for macOS

### PR DESCRIPTION
**Pull request by maksimdrachov**
This should address my questions in #248 

I might have some additional notes/questions, so don't merge just yet.

### CODE REVIEW

**Comment by maksimdrachov** [comment]
Ok, the demo is working for me now.

There appears to be some issue with the docs?

File: docs/pages/demo.rst [hunk 1]
```diff
@@ -72,6 +72,11 @@ You should end up with the following directory structure::
         ...
     demo_app.py                                     # The thermostat node script
 
+``CYPHAL_PATH`` should contain a list to all the paths where the DSDL root namespace directories are to be found:
+
+..  code-block:: sh
+    export CYPHAL_PATH="$HOME/pycyphal-demo/custom_data_types:$HOME/pycyphal-demo/public_regulated_data_types"
```
Comment (pavel-kirienko) [inline docs/pages/demo.rst:78]: The docs build failed because of the missing newline here.

```suggestion

    export CYPHAL_PATH="$HOME/pycyphal-demo/custom_data_types:$HOME/pycyphal-demo/public_regulated_data_types"
```

File: docs/pages/demo.rst [hunk 2]
```diff
@@ -72,6 +72,11 @@ You should end up with the following directory structure::
         ...
     demo_app.py                                     # The thermostat node script
 
+``CYPHAL_PATH`` should contain a list to all the paths where the DSDL root namespace directories are to be found:
```
Comment (pavel-kirienko) [inline docs/pages/demo.rst:75]:
```suggestion
``CYPHAL_PATH`` should contain a list of all the paths where the DSDL root namespace directories are to be found
(be sure to modify the value to match your environment):
```

File: docs/pages/demo.rst [hunk 3]
```diff
@@ -377,7 +385,16 @@ indicating that the thermostat is driving the plant towards the setpoint:
       kelvin: 302.1288757324219
     # And so on. Notice how the temperature is rising slowly towards the setpoint at 450 K!
 
-You can run ``yakut monitor`` to see what is happening on the network.
+You can run ``yakut monitor`` to see what is happening on the network. (Don't forget to set ``UAVCAN__UDP__IFACE``)
```
Comment (pavel-kirienko) [inline docs/pages/demo.rst:388]:
```suggestion
You can run ``yakut monitor`` to see what is happening on the network.
(Don't forget to set ``UAVCAN__UDP__IFACE`` or similar depending on your transport.)
```

File: docs/pages/demo.rst [hunk 4]
```diff
@@ -377,7 +385,16 @@ indicating that the thermostat is driving the plant towards the setpoint:
...
+
+.. tip:: macOS
+
+    Monitoring the network using ``yakut monitor``, requires using root while preserving your environment variables:
+
+    .. code-block:: sh
+
+    sudo -E yakut monitor
```
Comment (pavel-kirienko) [inline docs/pages/demo.rst:396]: Do you not need an extra level of indentation here for the code block to render properly?

File: docs/pages/synth/installation_option_matrix.py [hunk 5]
```diff
@@ -18,7 +18,8 @@
 
 
 print("If you need full-featured library, use this and read no more::", end="\n\n")
-print(f'   pip install pycyphal[{",".join(extras.keys())}]', end="\n\n")
+print(f'   pip install pycyphal[{",".join(extras.keys())}] # Linux', end="\n\n")
+print(f'   pip install pycyphal[{",".join(extras.keys())}] # macOS', end="\n\n")
```
Comment (pavel-kirienko) [inline docs/pages/synth/installation_option_matrix.py:22]: What is the purpose of this change? Also, it should be GNU/Linux, not Linux.

**Review by pavel-kirienko — CHANGES_REQUESTED** [review]

File: docs/pages/synth/installation_option_matrix.py (hunk 5 above)
Comment (maksimdrachov) [reply docs/pages/synth/installation_option_matrix.py:22]: Just showing the command is slightly different when using GNU/Linux vs macOS.

Comment (maksimdrachov) [reply docs/pages/synth/installation_option_matrix.py:22]: Ah I forgot to put the right command for macOS...

File: docs/pages/demo.rst (hunk 4 above)
Comment (maksimdrachov) [reply docs/pages/demo.rst]: Added

File: docs/pages/synth/installation_option_matrix.py (hunk 5 above)
Comment (pavel-kirienko) [reply docs/pages/synth/installation_option_matrix.py]: I see. The quotes are not really specific to macOS, they are required for zsh, which may be encountered in other systems as well (I am using zsh with GNU/Linux, for instance). I suggest modifying this by simply adding the quotes to the original command:

```diff
-print(f'   pip install pycyphal[{",".join(extras.keys())}]', end="\n\n")
+print(f'   pip install \'pycyphal[{",".join(extras.keys())}]\'', end="\n\n")
```

**Comment by pavel-kirienko** [comment]
Please see my comment above; also, the [doc8 test is failing](https://ci.appveyor.com/project/Zubax/pycyphal/builds/44945198/job/oyybokdrb5gyjm4c#L786):

> docs/pages/demo.rst:390: D002 Trailing whitespace

**Review by pavel-kirienko — APPROVED** [review]

**Comment by maksimdrachov** [comment]
What's the issue?

error

**Comment by pavel-kirienko** [comment]
@maksimdrachov please try disabling the doc8 check (in the noxfile.py) for all Python versions except the latest. This issue is not related to your work.

@silverv cc

**Comment by silverv** [comment]
I am trying to disable it now
